src/main/groovy/net/minecraftforge/gradleutils/GradleUtils.groovy

```diff
@@ -64,34 +79,118 @@ class GradleUtils {
+
+            return current.layout.projectDirectory
+        }
+    }
+
+    /**
+     * Find the {@code .git} directory of the Git repository the given project might be in.
+     *
+     * @param project The project to find the Git directory for
+     * @return A provider that resolves to the Git directory
+     *
+     * @see #findGitRoot(Project)
+     */
+    static Provider<Directory> findGitDirectory(Project project) {
+        return findGitRoot(project).map { it.dir(".git") }
+    }
+
+    static String makeFilterFromSubproject(Project project) {
+        var root = project.rootProject.projectDir
+        var local = project.projectDir
+
+        var result = local.absolutePath.substring(root.absolutePath.length())
+        if (result.startsWith(File.separator))
+            result = result.substring(1)
+        return result
+    }
+
+    // I LOVE NOT BREAKING BINARY COMPAT!!!!!! :) :) :)
+    static Map<String, String> gitInfoCheckSubproject(Project project, String filterFromSubproject) {
+        return gitInfo(findGitRoot(project).get().asFile, (git, tag) -> filterFromSubproject ? getSubprojectCommitCount(git, tag, filterFromSubproject) : null, filterFromSubproject, new String[0])
+    }
+
+    // I LOVE NOT BREAKING BINARY COMPAT!!!!!! :) :) :)
+    static Map<String, String> gitInfoCheckSubproject(Project project, String filterFromSubproject, String tagPrefixOverride) {
+        return gitInfo(findGitRoot(project).get().asFile, (git, tag) -> filterFromSubproject ? getSubprojectCommitCount(git, tag, filterFromSubproject) : null, tagPrefixOverride, new String[0])
+    }
+
+    // I LOVE NOT BREAKING BINARY COMPAT!!!!!! :) :) :)
+    static Map<String, String> gitInfoCheckSubproject(Project project, String filterFromSubproject, String tagPrefixOverride, String globFilter) {
+        return gitInfo(findGitRoot(project).get().asFile, (git, tag) -> filterFromSubproject ? getSubprojectCommitCount(git, tag, filterFromSubproject) : null, tagPrefixOverride, new String[] { globFilter })
+    }
+
+    static Map<String, String> gitInfo(Project project, String... globFilters) {
+        return gitInfo(findGitRoot(project).get().asFile, (git, tag) -> getSubprojectCommitCount(git, tag, makeFilterFromSubproject(project)), globFilters)
+    }
+
+    @Deprecated(forRemoval = true, since = "2.3")
     static Map<String, String> gitInfo(File dir, String... globFilters) {
+        return gitInfo(dir, (git, tag) -> null, null, globFilters)
+    }
+
+    static Map<String, String> gitInfo(File dir, BiFunction<Git, String, @Nullable Integer> commitCountProvider, String... globFilters) {
+        return gitInfo(dir, commitCountProvider, null, globFilters)
+    }
+
+    static Map<String, String> gitInfo(File dir, BiFunction<Git, String, @Nullable Integer> commitCountProvider, @Nullable String tagPrefix, String... globFilters) {
+        println "Getting git info! Dir: $dir, tagPrefix: $tagPrefix, globFilters: [${String.join(', ', globFilters)}]"
```

Comment:
Debug line needs to go away or be logged as debug so as to not showup in normal builds.